src/pybind/mgr/cephadm/module.py

```diff
@@ -901,14 +901,21 @@ def available(self) -> Tuple[bool, str, Dict[str, Any]]:
     def _validate_and_set_ssh_val(self, what: str, new: Optional[str], old: Optional[str]) -> None:
         self.set_store(what, new)
         self.ssh._reconfig_ssh()
-        if self.cache.get_hosts():
+        reachable_hosts = [h for h in self.cache.get_hosts() if h not in [
+            uh.hostname for uh in self.cache.get_unreachable_hosts()]]
```

Comment:
for readability I think it's better to have `[uh.hostname for uh in self.cache.get_unreachable_hosts()]` on its own variable `unreachable_hosts` for example.